sdk/textanalytics/ai-text-analytics/src/generated/generatedClientContext.ts

```diff
@@ -38,8 +38,11 @@ export class GeneratedClientContext extends coreHttp.ServiceClient {
 
     super(undefined, options);
 
-    this.baseUri = "{Endpoint}/text/analytics/v3.0";
     this.requestContentType = "application/json; charset=utf-8";
+
+    this.baseUri = options.endpoint || "{Endpoint}/text/analytics/v3.0";
```

Comment:
@joheredi, @sarangan12,  

What is the difference between the `endpoint` passed to the constructor and the `endpoint` in the `options` passed to the constructor?

Reply:
`options.endpoint` is added to all generated clients to allow consumers to override the default baseUrl specified in the swagger. This optional endpoint parameter doesn't come from the SWAGGER.

constructor's `endpoint` is defined in the Swagger as a Client parameter and dynamically generated.

In this case the intent is, if nothing is passed in `options.endpoint` the baseUri is a parametrized url `{Endpoint}/text/analytics/v3.0"`, core-http ([here](https://github.com/Azure/azure-sdk-for-js/blob/5cd418739507d4833f6c1ea5bfef2ae999f10a98/sdk/core/core-http/src/serviceClient.ts#L325)) will then replace `{Endpoint}` serviceClient's endpoint property (which is assigned with the value passed in the constructor).

It is probably weird to have `this.baseUri = options.endpoint` maybe it will be better to rename it to baseUrl  `this.baseUrli = options.baseUrl`, filed Azure/autorest.typescript/issues/702